src/main/primitives/Matrix3x3.java

```diff
@@ -0,0 +1,170 @@
+package primitives;
+
+import util.NormalCompare;
+
+/**
+ * The {@link Vector} class represents a {@link Vector} with it's base at the origin and it's head at the {@link Point}
+ * 'head'.
+ *
+ * @author Abraham Murciano
+ * @author Eli Levin
+ */
+public class Matrix3x3 {
+
+	// rows of matrix
+	public Vector r1;
+	public Vector r2;
+	public Vector r3;
+	public static final Matrix3x3 IDENTITY =
+			new Matrix3x3(new Vector(1, 0, 0), new Vector(0, 1, 0), new Vector(0, 0, 1));
+
+	public Matrix3x3(Vector r1, Vector r2, Vector r3) {
+		this.r1 = r1;
+		this.r2 = r2;
+		this.r3 = r3;
+	}
+
+	public Matrix3x3(Matrix3x3 A) {
+		this(A.r1, A.r2, A.r3);
+	}
+
+	public Matrix3x3 scale(double t) {
+		r1 = r1.scale(t);
+		r2 = r2.scale(t);
+		r3 = r3.scale(t);
+		return this;
+	}
+
+	private Triple multiply_T(Triple t) {
+		try {
+			Vector temp = new Vector(t);
+			return new Triple(r1.dot(temp), r2.dot(temp), r3.dot(temp));
+		} catch (ZeroVectorException e) {
+			return new Triple(0, 0, 0);
+		}
+	}
+
+	public Point multiply(Point p) {
+		return new Point(multiply_T(p));
+	}
+
+	public Vector multiply(Vector v) {
+		return new Vector(multiply_T(v));
+	}
+
+	public Matrix3x3 multiply(Matrix3x3 B) {
+		Vector col1 = new Vector(B.r1.x, B.r2.x, B.r3.x);
+		Vector col2 = new Vector(B.r1.y, B.r2.y, B.r3.y);
+		Vector col3 = new Vector(B.r1.z, B.r2.z, B.r3.z);
+		return new Matrix3x3(new Vector(r1.dot(col1), r1.dot(col2), r1.dot(col3)),
+				new Vector(r2.dot(col1), r2.dot(col2), r2.dot(col3)),
+				new Vector(r3.dot(col1), r3.dot(col2), r3.dot(col3)));
+	}
+
+
+	// After much keyboard smashing and tears (and too much math), this somehow works.
+	// Still needs testing tho :o
+	//
+	// Supposedly, you give it 2 vectors and it will return a rotation matrix that makes
+	// the first vector land on the second vector.
+	// feat(https://en.wikipedia.org/wiki/Rotation_matrix)
```

Comment:
phew... i thought for a second this was a javadoc comment...

Reply:
I can't decide if I should delete Matrix Class. I don't think anyone uses it anymore. It could be useful, and it was fun to make.

Reply:
Then just delete it. You can keep a copy of you really want to